docs/source/fetching-queries.md

```diff
@@ -27,16 +27,35 @@ query HeroName($episode: Episode) {
 Apollo iOS will generate a `HeroNameQuery` class that you can construct (with variables) and pass to `ApolloClient#fetch(query:)`:
 
 ```swift
-apollo.fetch(query: HeroNameQuery(episode: .empire)) { (result, error) in
-  print(result?.data?.hero?.name) // Luke Skywalker
+apollo.fetch(query: HeroNameQuery(episode: .empire)) { result in
+  if let name = try? result.get().data?.hero?.name {
+    print(name) // Luke Skywalker
+  }
 }
 ```
 
 > By default, Apollo will deliver query results on the main thread, which is probably what you want if you're using them to update the UI. `fetch(query:)` takes an optional `queue:` parameter however, if you want your result handler to be called on a background queue.
 
-The `error` parameter to the completion handler signals network or response format errors (such as invalid JSON).
+To handle potential errors, check the `failure(Failure)` result case, which details network or response format errors (such as invalid JSON):
+
+```swift
+apollo.fetch(query: HeroNameQuery(episode: .empire)) { result in
+  switch result {
+  case .success(let graphQLResult):
+    if let name = graphQLResult.data?.hero?.name {
+      print(name) // Luke Skywalker
+    } else if let errors = graphQLResult.errors {
+      // GraphQL errors
+      print(errors)
+    }
+  case .failure(let error):
+    // Network or response format errors
+    print(error)
+  }
+}
+```
 
-In addition to an optional `data` property, `result` contains an optional `errors` array with GraphQL errors (for more on this, see the sections on [error handling](https://facebook.github.io/graphql/#sec-Error-handling) and the [response format](https://facebook.github.io/graphql/#sec-Response-Format) in the GraphQL specification).
+In addition to an optional `data` property, `success(Success)` result case contains an optional `errors` array with GraphQL errors (for more on this, see the sections on [response format errors](https://graphql.github.io/graphql-spec/June2018/#sec-Errors) in the GraphQL specification).
```

Comment:
This link may need to be updated in the future, but the past one redirects to the spec home page which is less useful.